lib/models/cell.dart

```diff
@@ -0,0 +1,41 @@
+import 'dart:core';
+import 'package:andax/models/cell_write.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+part 'cell.g.dart';
+
+enum CellDisplay { check, text, range }
+
+@JsonSerializable()
+class Cell {
+  Cell(
+    this.id, {
+    this.max,
+    this.display,
+  });
+
+  final String id;
+  CellDisplay? display;
+  int? max;
+
+  String value = '';
+
+  void apply(CellWrite write) {
+    if (write.mode == CellWriteMode.overwrite) {
+      value = write.value;
+      return;
+    }
+
+    final oldNum = int.tryParse(value) ?? 0;
+    var num = int.tryParse(write.value) ?? 0;
```

Comment:
What do you mean by "robustness"? IMHO, less thrown errors do not mean that the app is more robust, it just means that errors (or invalid states) are ignored silently, which may cause problems in the future. This is the main issue with JavaScript and how it has weird behaviors allowing for so much wrong code that still doesn't fail.
I prefer preventing the problems at their source and disallowing any invalid input. If the user enters `one` and then finds no errors but also finds that the value was understood as `0`, then something is not right.

Reply:
Here I mean that the same cell can switch between numeric and text value depending on what designer wants. And I see no point in throwing error here because the value is actually a string anyway.